tortoise/state.go

```diff
@@ -202,7 +209,11 @@ func (s *state) Evict(ctx context.Context, windowStart types.LayerID) error {
 		}
 		batch.Reset()
 	}
+	oldEpoch := s.LastEvicted.GetEpoch()
 	s.LastEvicted = windowStart.Sub(1)
+	if s.LastEvicted.GetEpoch() > oldEpoch {
```

Comment:
i think we need to evict only if last evicted is the last layer in the epoch, otherwise cache will be invalidated too early

Reply:
i ended up using a map to collect epoch to evict in case we evicted multiple epochs at a time.